lib-multisrc/libgroup/src/eu/kanade/tachiyomi/multisrc/libgroup/LibGroup.kt

```diff
@@ -18,284 +22,213 @@ import eu.kanade.tachiyomi.source.model.Page
-                latestUpdatesParse(response)
+    @SuppressLint("SetJavaScriptEnabled")
+    @Suppress("NAME_SHADOWING")
+    private fun getToken() {
+        val latch = CountDownLatch(1)
+        Handler(Looper.getMainLooper()).post {
+            val webView = WebView(Injekt.get<Application>())
+            with(webView.settings) {
+                javaScriptEnabled = true
+                domStorageEnabled = true
+                databaseEnabled = true
+            }
+            webView.webViewClient = object : WebViewClient() {
+                override fun onPageFinished(view: WebView?, url: String?) {
+                    val view = view!!
+                    val script = "javascript:localStorage['auth']"
+                    view.evaluateJavascript(script) {
+                        view.stopLoading()
+                        view.destroy()
+                        if (!it.isNullOrBlank() && it != "null") {
+                            val str: String = if (it.first() == '"' && it.last() == '"') {
+                                it.substringAfter("\"").substringBeforeLast("\"")
+                                    .replace("\\", "")
+                            } else {
+                                it.replace("\\", "")
+                            }
+                            val token = str.parseAs<JsonObject>().jsonObject["token"]
+                            bearerToken =
+                                token!!.jsonObject["token_type"]!!.jsonPrimitive.content + " " + token.jsonObject["access_token"]!!.jsonPrimitive.content
+                        } else {
+                            bearerToken = "none"
+                        }
+                        latch.countDown()
+                    }
+                }
             }
+            webView.loadUrl(baseUrl)
+        }
+        latch.await(20, TimeUnit.SECONDS)
     }
 
-    override fun latestUpdatesParse(response: Response) =
-        popularMangaParse(response)
+    override fun getMangaUrl(manga: SManga): String {
+        return "$baseUrl/ru/manga${manga.url}"
+    }
 
-    // Popular
-    override fun popularMangaRequest(page: Int) = GET(baseUrl, headers)
-    override fun fetchPopularManga(page: Int): Observable<MangasPage> {
-        if (csrfToken.isEmpty()) {
-            return client.newCall(popularMangaRequest(page))
-                .asObservableSuccess()
-                .flatMap { response ->
-                    // Obtain token
-                    val resBody = response.body.string()
-                    csrfToken = "_token\" content=\"(.*)\"".toRegex().find(resBody)!!.groups[1]!!.value
-                    return@flatMap fetchPopularMangaFromApi(page)
-                }
-        }
-        return fetchPopularMangaFromApi(page)
+    // Latest
+    override fun latestUpdatesRequest(page: Int): Request {
+        val url = "https://api.$apiDomain/api/latest-updates".toHttpUrl().newBuilder()
+            .addQueryParameter("page", page.toString())
```

Comment:
site id not needed here?

Reply:
Not needed in params because already is in headers. 
I think api server has 2 ways to know for what site call is: url params and header field.
I added `site_id` in headers and where site itself uses it as url param, added it too. To be safe and use same network logic as they're site.